Directory.Build.props

```diff
@@ -23,6 +23,10 @@
     <RollForward>Major</RollForward>
   </PropertyGroup>
 
+  <PropertyGroup Condition=" '$(MSBuildRuntimeType)' == 'Core' ">
```

Comment:
Is building with `dotnet build` a requirement for this one? When you build inside Visual Studio, it will be `Full` using .NET Framework MSBuild.

Reply:
Yep, marshal methods will be NET7+ only

Reply:
Well, but you can build .NET 7 projects in Visual Studio with .NET Framework MSBuild.

Does this feature require our build to be running under .NET 7?

Reply:
We can maybe shell out to `dotnet` to make this work, if needed.

Reply:
Native portion of the feature uses APIs that aren't available in Mono classic, so I wanted to make sure it's not enabled for classic by accident. However, this properly is here just until I finish all the parts of the feature, it will be
removed in the last PR. It's just easier this way for me to keep the feature disabled in `main` while I can enable it globally in the branch I'm working in on the next PRs.